vignettes/secuTrialR-package-vignette.Rmd

```diff
@@ -1,172 +1,568 @@
-are followed by data frames containing meta data tables and finally the central study data tables.
-At this point you can also see how exports with long names and short names differ.
+Since the centre names often have a systematic addition (e.g. (RPACK)) we have
+enabled the option to remove certain parts of the centre descriptions via
+regular expressions (i.e. `rm_regex` argument). In this case the regular expression
+removes trailing parentheses and everything they enclose.
+\vspace{5pt}
 
 ```{r}
+annual_recruitment(ctu05_data, rm_regex = "\\(.*\\)$")
+```
+
+\newpage
+
+It is also possible to plot the recruitment over time.
+\vspace{5pt}
+
+```{r, fig.height = 3.6, fig.width = 8}
+plot_recruitment(ctu05_data, cex = 1.2, rm_regex = "\\(.*\\)$")
+```
+
+## Visit plan visualization
+
+`secuTrialR` can provide a depiction of the visit structure, although only where
+the visit plan is fixed.
+\vspace{5pt}
+
+```{r, fig.height = 4, fig.width = 4}
+vs <- visit_structure(ctu05_data)
+plot(vs)
+```
+
+## Completeness of forms
 
-names(sT_export_shortnames)
-names(sT_export_longtnames)
+If you are not sure about how complete the data in your export is, it may be useful to
+get a quick overview of how well the forms have been filled. The below table shows
+both absolute and relative numbers for a few forms.
+\vspace{5pt}
 
+```{r}
+fss <- form_status_summary(ctu05_data)
+tail(fss, n = 5)
 ```
 
-The study data for the bmd secuTrial form is stored in bmd for the short table names and dem00bmd for the long table names.
-As expected they are the same if compared, since the only difference is the formatting of the table names.
+Please note that a form is only complete if all required fields have been filled.
+Thus, a whole study may have 99% completeness on variable basis while showing
+0% completeness on form basis. It is currently not technically possible to assess
+completeness on variable basis in a generic way. Hence, high completeness on form
+basis implies high completeness on variable basis but **NOT** vice versa.
+
+For a more participant id centred statistic you can perform the following.
+\vspace{5pt}
 
 ```{r}
-head(sT_export_shortnames$bmd[,c("pat_id", "age", "grouping", "bmd")], n = 4)
+fsc <- form_status_counts(ctu05_data)
+# show the top
+head(fsc)
+```
+
+## Form linkage
 
-all.equal(sT_export_shortnames$bmd, sT_export_longtnames$dem00bmd)
+Linkages amongst forms can be explored with the `links_secuTrial()` function.
+This relies on the `igraph` package to create a network. It is possible to
+interact with the network, e.g. move nodes around in order to read the labels better.
+The device ID is returned to the console, but can be ignored. Forms are plotted in
+deep yellow, variables in light blue.
+\vspace{5pt}
 
+```{r, eval = FALSE}
+links_secuTrial(ctu05_data)
 ```
 
+The output can not be shown within this vignette but you should give it a try.
```

Comment:
Maybe make a comment that with audit trail exported, there are too many links for it to be useful...?